.github/workflows/build_deploy.yml

```diff
@@ -27,7 +27,7 @@ jobs:
       contents: read
     steps:
       - name: Check out the repo
-        uses: actions/checkout@v3
+        uses: actions/checkout@4
```

Comment:
The use of `actions/checkout@4` in the other Actions should validate this change.